lib/rest-api-stuff.php

```diff
@@ -46,6 +46,11 @@ function add_rest_api_endpoints() {
 			$features['woocommerce'] = $json_params['woocommerce'];
 		}
 
+		if ( isset( $json_params['jetpack-beta'] ) ) {
+			$features['jetpack-beta'] = $json_params['jetpack-beta'];
+			$features['branch'] = $json_params['branch'];
```

Comment:
Let's add somewhere around here the check for a valid branch fetching the manifest with wp_remote_request() and a sort of try/catch around it fi possible.